### DISCUSSION TITLE
Is -compress working or broken? Because the file size remained the same.

### DISCUSSION THREAD

**Discussion by karl-police** [ImageMagick/ImageMagick#3748] [Help]
Is -compress working or broken? Because the file size remained the same.

**Comment by snibgo** [comment]
Your question is vague. What version of IM, on what platform? Please show your command(s).

Yes, "-compress" works. Not all "-compress" methods are available for all file formats.

**Reply by karl-police** [reply]
> Your question is vague. What version of IM, on what platform? Please show your command(s).
> 
> Yes, "-compress" works. Not all "-compress" methods are available for all file formats.

Is there one for GIF?

**Comment by snibgo** [comment]
As far as I know, "-compress" has no effect when writing GIF files.

**Comment by snibgo** [comment]
EDIT: I should add that IM does use compression on GIFs, but the user has no control over the method.

**Reply by karl-police** [reply]
> EDIT: I should add that IM does use compression on GIFs, but the user has no control over the method.

IM?

**Reply by fmw42** [reply]
Actually there is some control using -treedepth, if I recall correctly. See https://imagemagick.org/script/command-line-options.php#treedepth

**Reply by karl-police** [reply]
> Actually there is some control using -treedepth, if I recall correctly. See https://imagemagick.org/script/command-line-options.php#treedepth

``-treedepth 0`` ?

**Reply by snibgo** [reply]
Thanks, fmw42. That may be true, but I can't find an example where "-treedepth N" makes a difference to the size of a written GIF.

**Reply by fmw42** [reply]
Yes, that was my evaluation also. But I thought I would mention that option.

**Reply by karl-police** [reply]
I didn't notice a difference. What I know about GIFs is that there are local color tables or a global color table. Not sure if ImageMagick has something that controls that.

**Comment by snibgo** [comment]
IM = ImageMagick.
